Dockerfile

```diff
@@ -60,17 +86,31 @@ RUN apk add -u shadow libc6-compat && \
 
 USER secrets-provider
 
-COPY --from=secrets-provider-builder /opt/secrets-provider-for-k8s/secrets-provider /usr/local/bin/
+# =================== RELEASE MAIN CONTAINER ===================
+FROM secrets-provider-base as secrets-provider
 
-ENTRYPOINT [ "/usr/local/bin/secrets-provider"]
+COPY --from=secrets-provider-builder /opt/secrets-provider/secrets-provider /usr/local/bin/
+
+CMD [ "/usr/local/bin/secrets-provider"]
+
+# =================== DEBUG MAIN CONTAINER ===================
+FROM secrets-provider-base as secrets-provider-debug
+
+COPY --from=secrets-provider-builder-debug /go/bin/dlv /usr/local/bin/
+
+COPY --from=secrets-provider-builder-debug /opt/secrets-provider/secrets-provider /usr/local/bin/
+
+# Execute secrets provider wrapped with dlv debugger listening on port 40000 for remote debugger connection.
+# Will wait indefinitely until a debugger is connected.
+CMD ["/usr/local/bin/dlv", "--listen=:40000", "--headless=true", "--api-version=2", "--accept-multiclient", "exec", "/usr/local/bin/secrets-provider"]
 
 # =================== MAIN CONTAINER (REDHAT) ===================
-FROM registry.access.redhat.com/rhel as secrets-provider-for-k8s-redhat
+FROM registry.access.redhat.com/rhel as secrets-provider-redhat
 MAINTAINER CyberArk Software Ltd.
 
 ARG VERSION
 
-LABEL name="secrets-provider-for-k8s"
+LABEL name="secrets-provider"
```

Comment:
i wouldn't change the label as it is a breaking change. I am not sure but the label may be present on the RH site and customers may be looking for it.

i see that you chose to go with `secrets-provider` instead of `secrets-provider-for-k8s` throughout the file. maybe the latter is better.

Reply:
I'm against this change, this was not my decision to call it like that, and all the file is full of `secrets-provider`.
I'll revert it.

Reply:
i think it is also better to use the full name anyway. sorry for not saying so earlier.

Reply:
again, sorry i didn't decide earlier on my preffered name but I do think that the full name is better. So will you change all the occurrences of `secrets-provider` to `secrets-provider-for-k8s` throughout the code? I think it is better to have uniformity.

Reply:
I added it to the dockerfile improvements issue I opened: #150